src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl implements RoleDao {
+    private final SessionFactory factory;
+
+    public RoleDaoImpl(SessionFactory factory) {
+        this.factory = factory;
+    }
+
+    @Override
+    public void add(Role role) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(role);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert role " + role, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<Role> getRoleByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            Query<Role> findByRoleName = session.createQuery(
+                    "FROM Role WHERE roleName = :roleName", Role.class);
+            findByRoleName.setParameter("roleName", Role.RoleName.valueOf(roleName));
+            return findByRoleName.uniqueResultOptional();
+        } catch (Exception e) {
+            throw new DataProcessingException("Role with roleName " + roleName + " not found", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Cant get Role by RoleName" + roleName,  e);
```

We get exception here if smth went wrong while we getting Role from db